crates/bevy_text/src/pipeline.rs

```diff
@@ -107,6 +107,9 @@ impl TextPipeline {
         computed.entities.clear();
 
         for (span_index, (entity, depth, span, text_font, color)) in text_spans.enumerate() {
+            if span.is_empty() {
+                continue;
+            }
```

Comment:
For correct indexing, this should go after `computed.entities.push(TextEntity { entity, depth });`.

Reply:
Ah yeah. I thought I tested this properly, but it was setting the wrong span and I didn't know because the colors were the same. Sure enough, giving the spans different colors surfaced an indexing problem, and the reorder fixes it.

Reply:
Nice catch!